openadapt/config.py

```diff
@@ -138,107 +37,293 @@
+
+    # Warning suppression
+    IGNORE_WARNINGS: bool = False
+    MAX_NUM_WARNINGS_PER_SECOND: int = 5
+    WARNING_SUPPRESSION_PERIOD: int = 1
+    MESSAGES_TO_FILTER: list[str] = ["Cannot pickle Objective-C objects"]
+
+    # Action event configurations
+    ACTION_TEXT_SEP: str = "-"
+    ACTION_TEXT_NAME_PREFIX: str = "<"
+    ACTION_TEXT_NAME_SUFFIX: str = ">"
+
+    # Performance plotting
+    PLOT_PERFORMANCE: bool = True
+
+    # App configurations
+    APP_DARK_MODE: bool = False
+
+    # Scrubbing configurations
+    SCRUB_ENABLED: bool = False
+    SCRUB_CHAR: str = "*"
+    SCRUB_LANGUAGE: str = "en"
+    SCRUB_FILL_COLOR: Union[str, int] = 0x0000FF
+
+    @field_validator("SCRUB_FILL_COLOR")
+    @classmethod
+    def validate_scrub_fill_color(cls, v: Union[str, int]) -> int:  # noqa: ANN102
+        """Validate the scrub fill color.
+
+        Convert the color to a hex value if it is a string.
+        """
+        if isinstance(v, str):
+            return int(v, 16)
+        return v
+
+    SCRUB_CONFIG_TRF: dict = {
+        "nlp_engine_name": "spacy",
+        "models": [{"lang_code": "en", "model_name": "en_core_web_trf"}],
+    }
+    SCRUB_PRESIDIO_IGNORE_ENTITIES: list[str] = []
+    SCRUB_KEYS_HTML: list[str] = [
+        "text",
+        "canonical_text",
+        "title",
+        "state",
+        "task_description",
+        "key_char",
+        "canonical_key_char",
+        "key_vk",
+        "children",
+    ]
+
+    # Visualization configurations
+    VISUALIZE_DARK_MODE: bool = False
+    VISUALIZE_RUN_NATIVELY: bool = True
+    VISUALIZE_DENSE_TREES: bool = True
+    VISUALIZE_ANIMATIONS: bool = True
+    VISUALIZE_EXPAND_ALL: bool = False
+    VISUALIZE_MAX_TABLE_CHILDREN: int = 10
+
+    # Screenshot configurations
+    SAVE_SCREENSHOT_DIFF: bool = False
+
+    # Spacy configurations
+    SPACY_MODEL_NAME: str = "en_core_web_trf"
+
+    # Dashboard configurations
+    DASHBOARD_CLIENT_PORT: int = 3000
+    DASHBOARD_SERVER_PORT: int = 8000
+
+    SOM_SERVER_URL: str = "<SOM_SERVER_URL>"
+
+    class Adapter(str, Enum):
+        """Adapter for the completions API."""
+
+        OPENAI: str = "openai"
+        ANTHROPIC: str = "anthropic"
+        GOOGLE: str = "google"
+
+    DEFAULT_ADAPTER: Adapter = Adapter.OPENAI
+
+    @classmethod
+    def settings_customise_sources(
+        cls,  # noqa: F821,ANN102
+        settings_cls: type[BaseSettings],
+        init_settings: PydanticBaseSettingsSource,
+        env_settings: PydanticBaseSettingsSource,
+        dotenv_settings: PydanticBaseSettingsSource,
+        file_secret_settings: PydanticBaseSettingsSource,
+    ) -> tuple[PydanticBaseSettingsSource, ...]:
+        """Define the sources and their order for loading the settings values.
+
+        The order of preference is as follows:
+        1. settings with which the class is initialized
+        2. settings from the local config json file
+        3. settings from the config json file
```

Comment:
Can you please clarify in the documentation the difference between the local config json file and the config json file?